### PR TITLE
Reworked output of phpcompat details

### PR DESCRIPTION
**Pull request by rheinardkorf**
```
{
  totals: {
    errors: 225,
    warnings: 0
  },
  error_map: {
    5.2: [
      PHPCompatibility.PHP.LateStaticBinding.Found
    ],
  },
  errors: {
    PHPCompatibility.PHP.LateStaticBinding.Found: {
      message: "Late static binding is not supported in PHP 5.2 or earlier.",
      source: "PHPCompatibility.PHP.LateStaticBinding.Found",
      type: "ERROR",
      severity: 5,
      versions: [
        "5.2"
      ],
      files: {
        includes/lib/Example.php: {
          line: 2199,
          column: 32
        }
      }
    }
  }
}
```

### CODE REVIEW

**Comment by rheinardkorf** [comment]
@valendesigns please note that I renamed `violation_map` and `violations` to `error_map` and `errors` respectively.  This is so that we can add warnings later too.

**Comment by valendesigns** [comment]
Aren’t there warnings in there, or is it only errors?

**Comment by valendesigns** [comment]
@rheinardkorf Are you leaving the warnings out because they don't fail the PHP version?

**Comment by rheinardkorf** [comment]
@valendesigns yup. We don't need them for the compat report, but it wouldn't hurt including them if you think its meaningful in a UI.

File: audit/phpcs/phpcompat.go [hunk 1]
```diff
@@ -77,26 +97,59 @@ func (p *PhpCompat) Process(msg message.Message, result *audit.Result) {
...
+					Files:    make(map[string][]FilePosition),
+				}
+
+				// Get incompatible versions
 				broken := phpcompat.BreaksVersions(sniffMessage)
+				violation.Versions = broken
+
+				// Add the source to each broken version.
+				for _, version := range broken {
+					details.ErrorMap[version] = append(details.ErrorMap[version], sniffMessage.Source)
```
Comment (valendesigns) [inline audit/phpcs/phpcompat.go:131]: Will this only append if unique?

**Review by valendesigns — APPROVED** [review]

**Pull request merged**